### PR TITLE
Properly testing pointer vars aren't null before using them.

### PR DESCRIPTION
**Pull request by Areloch**
Issue found by PVS Studio:

Several instances where we utilize a pointer variable without properly testing that they aren't null first.

### CODE REVIEW

**Comment by eightyeight** [comment]
Something that might have been worth investigating is whether any of these are _functionally_ never NULL, and in those cases whether passing a reference would be more appropriate.

**Comment by Areloch** [comment]
Aye, that'd probably be a good thing for a more in-depth cleanup.

That said, Az tested and noted that this is causing some weirdities with terrain rendering, so consider this NOT safe to merge in currently. I'll do more testing tomorrow and correct whatever's derping out.

File: Engine/source/renderInstance/renderPrePassMgr.cpp [hunk 1]
```diff
@@ -203,12 +203,11 @@ void RenderPrePassMgr::addElement( RenderInst *inst )
...
    // Skip decals if they don't have normal maps.
-   if ( isDecalMeshInst && !matInst->hasNormalMap() )
+   if (!matInst || isDecalMeshInst && !matInst->hasNormalMap())
       return;
 
    // If its a custom material and it refracts... skip it.
-   if (  matInst && 
-         matInst->isCustomMaterial() &&
+   if (  matInst->isCustomMaterial() &&
          static_cast<CustomMaterial*>( matInst->getMaterial() )->mRefract )
```
Comment (Azaezel) [inline Engine/source/renderInstance/renderPrePassMgr.cpp]: This block change looks to need more study. It's running afoul of the special snowflake terrain code.

File: Engine/source/renderInstance/renderPrePassMgr.cpp [hunk 2]
```diff
@@ -203,12 +203,11 @@ void RenderPrePassMgr::addElement( RenderInst *inst )
       matInst = static_cast<MeshRenderInst*>(inst)->matInst;
 
    // Skip decals if they don't have normal maps.
-   if ( isDecalMeshInst && !matInst->hasNormalMap() )
+   if (!matInst || isDecalMeshInst && !matInst->hasNormalMap())
```
Comment (eightyeight) [inline Engine/source/renderInstance/renderPrePassMgr.cpp]: Check order of operations here?

**Comment by Azaezel** [comment]
suggested revision after a few pokes:

   // Get the material if its a mesh.
   BaseMatInstance* matInst = NULL;
   if ( isMeshInst || isDecalMeshInst )
      matInst = static_cast<MeshRenderInst*>(inst)->matInst;

   if ( matInst )
   {
      // Skip decals if they don't have normal maps.
      if (isDecalMeshInst && !matInst->hasNormalMap())
         return;
      
      // If its a custom material and it refracts... skip it.
      if (matInst->isCustomMaterial() &&
         static_cast<CustomMaterial*>( matInst->getMaterial() )->mRefract )
         return;
      
      // Make sure we got a prepass material.      
      matInst = getPrePassMaterial( matInst );
      if ( !matInst || !matInst->isValid() )
         return;
   }

**Comment by Areloch** [comment]
Alright, PR updated, not seeing any problems on my end.

**Pull request merged**